### PR TITLE
Add dotnet/test-templates repo

### PR DESCRIPTION
**Pull request by crummel**

### CODE REVIEW

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -37,6 +37,11 @@
       <Uri>https://github.com/dotnet/templating</Uri>
       <Sha>096be35db97669a59b298168b2be30f9b43eab8a</Sha>
     </Dependency>
+    <!-- this should have a CPD attribute but this doesn't exist upstream yet.  see GitHub.com/issues/XXX -->
```
Comment (dagood) [inline eng/Version.Details.xml:40]: TODO

File: repos/test-templates.proj [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
+
+  <PropertyGroup>
+    <BuildCommandArgs>--restore --build --pack</BuildCommandArgs>
+    <BuildCommandArgs>$(BuildCommandArgs) --configuration $(Configuration)</BuildCommandArgs>
+    <BuildCommandArgs>$(BuildCommandArgs) --binaryLog</BuildCommandArgs>
+    <BuildCommandArgs>$(BuildCommandArgs) -ci</BuildCommandArgs>
+    <BuildCommandArgs>$(BuildCommandArgs) $(FlagParameterPrefix)nodereuse $(ArcadeFalseBoolBuildArg)</BuildCommandArgs>
+    <BuildCommand>$(ProjectDirectory)build$(ShellExtension) $(BuildCommandArgs)</BuildCommand>
```
Comment (dagood) [inline repos/test-templates.proj:11]: Looks like the file doesn't have proper permission in the repo. Or anything in eng/common! 😕

**Review by dseefeld — APPROVED** [review]

File: repos/test-templates.proj (hunk 2 above)
Comment (crummel) [reply repos/test-templates.proj:11]: Upgraded to a newer hash and patched the permissions.

File: eng/Version.Details.xml (hunk 1 above)
Comment (crummel) [reply eng/Version.Details.xml:40]: This has been added upstream, removed.

**Pull request merged**